### PR TITLE
Infrastructure Tests

### PR DESCRIPTION
**Pull request by eStarKee06**
tests: 
- firebase auth repo
- contact form repo

### CODE REVIEW

File: test/infrastructure/contact_form/contact_form_repository_test.dart [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    final contactPostUrl = Uri.parse('$baseUrl/v1/contact');
+    final contactPostHeader = {
+      'accept': '*/*',
+      'Content-Type': 'application/json',
+    };
+    String expectedBody = json.encode({
+      'email': tContactFormDto.email,
+      'title': tContactFormDto.subject,
+      'body': tContactFormDto.message
+    });
```
Comment (wizlif) [inline test/infrastructure/contact_form/contact_form_repository_test.dart:39]: Why not use the `toJson()` method ??

File: test/infrastructure/auth/firebase_auth_repository_test.dart [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+      phoneNumber: any(named: 'phoneNumber'),
+      verificationCompleted: any(named: 'verificationCompleted'),
+      verificationFailed: any(named: 'verificationFailed'),
+      codeSent: any(named: 'codeSent'),
+      codeAutoRetrievalTimeout: any(named: 'codeAutoRetrievalTimeout')));
+
+  return {
+    "mockFirebaseAuth": mockFirebaseAuth,
+    "mockVerifyPhoneNumber": mockVerifyPhoneNumber
+  };
```
[the remark this answers is not in the document]
Comment (eStarKee06) [reply test/infrastructure/auth/firebase_auth_repository_test.dart:46]: sure thing

**Review by Xazin — APPROVED** [review]

**Comment by Xazin** [comment]
I approved it, but I thought I'd show you how you can simplify your testing.
https://gist.github.com/Xazin/2304c49a2dd8d3e0773b3c545f4e32e5
Check this gist, it's my take on the contact_repository_test.dart.

What is different? I am not trying to mock the specific data or headers, and still 100% coverage.

**Comment by eStarKee06** [comment]
Thank you so much @Xazin 

The headers does seem negligible but I have a question about the body of the api call. In this particular example, the method sets the body's 'title' field to ContactForm's  'subject' field. Since this is newly introduced logic and is essential to the call's success, shouldn't we test this?

Also, I remember reading that I can merge once it's approved but I just want to make sure that it is okay to merge now?

**Comment by Xazin** [comment]
> Thank you so much @Xazin 
> 
> The headers does seem negligible but I have a question about the body of the api call. In this particular example, the method sets the body's 'title' field to ContactForm's  'subject' field. Since this is newly introduced logic and is essential to the call's success, shouldn't we test this?
> 
> Also, I remember reading that I can merge once it's approved but I just want to make sure that it is okay to merge now? 
> 

Yeah you can merge.

It is not that you shouldn't test the header and body, but most of the time I would do this with integration tests, since the purpose of ak integration test is to test across services. Such as the API call.

When unit testing we want the result of the input to be aligned with the expected output. You are mocking the response from the API call therefore even if the API contract changed, the test would succeed.

**Pull request merged**